lib/Site.js

```diff
@@ -286,15 +287,17 @@ Site.prototype.createPage = function (config) {
  */
 Site.prototype.collectAddressablePages = function () {
   const { pages } = this.siteConfig;
-  const addressableGlobs = pages.filter(page => page.glob).map(page => page.glob);
+  const addressableGlobs = pages.filter(page => page.glob);
   this.addressablePages = pages.filter(page => page.src);
-  const globPaths = walkSync(this.rootPath, {
-    directories: false,
-    globs: addressableGlobs,
-    ignore: [BOILERPLATE_FOLDER_NAME],
-  }).map(globPath => ({ src: globPath }));
+  const globPaths = addressableGlobs.reduce((globPages, addressableGlob) => {
+    return globPages.concat(walkSync(this.rootPath, {
+      directories: false,
+      globs: [addressableGlob.glob],
+      ignore: [BOILERPLATE_FOLDER_NAME],
+    }).map(globPath => ({ src: globPath, searchable: addressableGlob.searchable })));
+  }, []);
   // Add pages collected by walkSync without duplication
-  this.addressablePages = _.unionWith(this.addressablePages, globPaths,
+  this.addressablePages = _.unionWith(globPaths, this.addressablePages,
```

Comment:
Is there a reason why we swapped the order here?

Reply:
As you explained face-to-face, the scenario in concern was this: 

```json
    {
      "src": "index.md",
      "title": "Hello World"
    },
    {
      "glob": "index.md",
      "searchable": "no"
    },
```

This will not work if the order is not swapped.

Reply:
Hmm, `src` should have higher priority than `glob`.

Reply:
Hmm, `src` should have higher priority than `glob` as its more specific.

Reply:
> Hmm, src should have higher priority than glob as it's more specific.

@acjh agree with you on this. @damithc do you agree with this as well?

Reply:
I'm ok with src having higher priority than glob.
```
{
      "src": "index.md",
      "title": "Hello World"
    },
    {
      "glob": "index.md",
      "searchable": "no"
    },
```
In the case above, it looks like the `"searchable": "no"` should be applied because `src` (although of higher priority) doesn't specify the searchable property. i.e., _specified_ should take priority over _not specified_. But I'm ok this is too hard to implement.